Built new status labels

kwm14 opened the pull request:
## Done
- Built a new set of `status` buttons
  - `p-button--validated`
  - `p-button--in-progress`
  - `p-button--new`
  - `p-button--updated`
  - `p-button--depreacated`
- Added   `@include vf-p-status;` to _vanilla.scss file
- Created seperate example pages for each button rather them all being in the same file

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/
- Check new buttons look like the desired design and are unclickable
  - http://0.0.0.0:8101/examples/patterns/status/validated/
  - http://0.0.0.0:8101/examples/patterns/status/in-progress/
  - http://0.0.0.0:8101/examples/patterns/status/new/
  - http://0.0.0.0:8101/examples/patterns/status/updated/
  - http://0.0.0.0:8101/examples/patterns/status/deprecated/
- Check documentation page
  - https://vanilla-framework-canonical-web-and-design-pr-2416.run.demo.haus/patterns/labels/

## Details
Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/2412

## Screenshots
<img width="102" alt="Screenshot 2019-07-23 at 12 06 20" src="https://user-images.githubusercontent.com/17748020/61707428-637b9100-ad42-11e9-881f-707763982812.png">
<img width="114" alt="Screenshot 2019-07-23 at 12 06 09" src="https://user-images.githubusercontent.com/17748020/61707427-637b9100-ad42-11e9-911e-0682e6822669.png">
<img width="68" alt="Screenshot 2019-07-23 at 12 06 33" src="https://user-images.githubusercontent.com/17748020/61707429-637b9100-ad42-11e9-9f9e-c243b6d34e59.png">
<img width="97" alt="Screenshot 2019-07-23 at 12 06 56" src="https://user-images.githubusercontent.com/17748020/61707431-64142780-ad42-11e9-89b0-94ac0bac720f.png">
<img width="118" alt="Screenshot 2019-07-23 at 12 05 47" src="https://user-images.githubusercontent.com/17748020/61707425-637b9100-ad42-11e9-81ff-52b44131ffa6.png">

webteam-app commented:
Starting demo at: https://vanilla-framework-canonical-web-and-design-pr-2416.run.demo.haus/

anthonydillon commented on scss/_patterns_status.scss:
```diff
@@ -0,0 +1,61 @@
+}
+
+// Override local variables
+@mixin vf-p-button-validated {
```
I think `vf-p-status-validated` is a better name as they are not buttons.

anthonydillon commented on scss/_patterns_status.scss:
```diff
@@ -0,0 +1,61 @@
+  @include vf-p-button-deprecated;
+}
+
+%p-button--status {
```
This should just be `%vf-status`

kwm14 commented:
> I prefer the to make this a new pattern then piggyback off the buttons pattern. Also, this PR is missing documentation.

Sure thing I can update. In regards to docuemntation, I have a separate [issue](https://github.com/canonical-web-and-design/vanilla-framework/issues/2413) for this, unless you want me to add it to this PR?

anthonydillon commented:
It is much better to do the sass, examples and docs in the same PR so that it can all be reviewed at the same time. That is the main reason for having the docs in the codebase.

anthonydillon commented on scss/_patterns_status.scss:
```diff
@@ -0,0 +1,61 @@
+@mixin vf-p-button-validated {
+  .p-button--validated {
+    @extend %p-button--status;
+    background-color: #006b75;
```
You need set hex values to a variable before using. I would suggest setting the vars at the top of the main mixin (vf-p-status) so its easy to change in the future.

spencerbygraves commented:
@kwm14 @anthonydillon should we be calling these buttons?

deadlight commented:
Can we rename these "p-status--"?

kwm14 commented:
> This is looking much better but missing documentation.

I know I know. I'm not that fast 😆

kwm14 commented:
> Can we rename these "p-status--"?

Sure thing. Conversation with @spencerbygraves we've agreed that `'label` will be better than `status` as these are used in JAAS as tags.

JAAS - Tag labels
Vanilla - Status labels

anthonydillon commented:
Just missing some docs :-1:

kwm14 commented:
## Documentation
- Added docs page to list out labels
- Added link to new component status page (will build next) so will 404
- Added link to labels design spec (will add to Design repo) so will 404

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/patterns/labels/
- Review page 
- Make sure label examples are displayed

kwm14 commented:
@anthonydillon added documentation page ❤️

anthonydillon commented on docs/patterns/labels.md:
```diff
@@ -0,0 +1,64 @@
+
+<hr>
+
+Labels are static elements which you can apply to HTML elements to signify status, tags or any other information you find useful.
```
I don't think you need this part of the sentence "to HTML elements"

kwm14 replied:
Fixed 👍

anthonydillon commented on docs/patterns/labels.md:
```diff
@@ -0,0 +1,64 @@
+
+<a href="/examples/patterns/labels/in-progress/"
+    class="js-example">
+View example of the labels pattern
```
Can the change to the name of the pattern for a11y

kwm14 replied:
Fixed 👍

kwm14 commented:
@spencerbygraves ready for design review 👍
